crates/move-prover-boogie-backend/src/generator.rs

```diff
@@ -309,10 +314,167 @@ fn generate_module_bpl<W: WriteColor>(
+
+        if excluded_addresses.contains(module_addr)
+            || move_model::model::GlobalEnv::SPECS_MODULES_NAMES.contains(&module_name.as_str())
+        {
+            continue;
+        }
+
+        for func_env in module_env.get_functions() {
+            if should_include_function(&func_env, targets) {
+                let module_name_str = func_env
+                    .module_env
+                    .get_name()
+                    .display(env.symbol_pool())
+                    .to_string();
+                functions_by_module
+                    .entry(module_name_str)
+                    .or_default()
+                    .push(func_env.get_qualified_id());
+            }
+        }
+    }
+
+    for (module_name, func_ids) in functions_by_module {
+        println!("📦 Module: {}", module_name);
+
+        if func_ids.is_empty() {
+            println!("  (no public functions)");
+        } else {
+            for func_id in func_ids {
+                let func_env = env.get_function(func_id);
+                total_public_functions += 1;
+
+                let status = determine_spec_status(&func_env, targets);
+                *stats_by_status.entry(format!("{}", status)).or_insert(0) += 1;
+
+                println!("  {} {}", status, func_env.get_name_str());
+            }
+        }
+        println!();
+    }
+
+    println!("📈 Summary:");
+    println!("Total public functions: {}", total_public_functions);
+    for (status, count) in stats_by_status {
+        println!("  {}: {}", status, count);
+    }
+}
+
+/// Determines the proof status of a function.
+///
+/// This function first attempts to find a spec for the given function using `get_spec_by_fun`.
+/// This is used when the function has an explicit spec mapping (e.g., for inlined or synthetic functions).
+/// If no spec is found this way, it falls back to checking if the function is present in the left side of
+/// `function_specs()`, which covers functions that have specs but are not mapped via `get_spec_by_fun`.
+///
+/// The distinction is important because some functions may have their specs mapped differently,
+/// and we want to ensure that all cases are covered when determining proof status.
+/// The function then checks if the spec is skipped, successfully verified, or present but not proved,
+/// returning the appropriate `ProofStatus` variant. If no spec is found at all, it returns `NoSpec`.
+fn determine_spec_status(
+    func_env: &move_model::model::FunctionEnv,
+    targets: &FunctionTargetsHolder,
+) -> ProofStatus {
+    let func_id = func_env.get_qualified_id();
+    let skip_specs_list: Vec<_> = targets.skip_specs().collect();
+
+    if let Some(spec_id) = targets.get_spec_by_fun(&func_id) {
+        if skip_specs_list.contains(&spec_id) {
+            ProofStatus::Skipped
+        } else if targets.ignores_aborts(spec_id) {
+            ProofStatus::IgnoreAborts
+        } else {
+            ProofStatus::SuccessfulProof
+        }
```

Comment:
we should distinguish between proved specs and specs that aren't proved (just `#[spec]`)